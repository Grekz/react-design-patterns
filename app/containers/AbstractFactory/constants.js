/*
 *
 * AbstractFactory constants
 *
 */

export const DEFAULT_ACTION = 'app/AbstractFactory/DEFAULT_ACTION';
