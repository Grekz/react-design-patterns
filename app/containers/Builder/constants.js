/*
 *
 * Builder constants
 *
 */

export const DEFAULT_ACTION = 'app/Builder/DEFAULT_ACTION';
