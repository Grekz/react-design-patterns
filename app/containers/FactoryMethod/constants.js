/*
 *
 * FactoryMethod constants
 *
 */

export const DEFAULT_ACTION = 'app/FactoryMethod/DEFAULT_ACTION';
