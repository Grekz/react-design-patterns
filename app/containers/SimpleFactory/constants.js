/*
 *
 * SimpleFactory constants
 *
 */

export const DEFAULT_ACTION = 'app/SimpleFactory/DEFAULT_ACTION';
