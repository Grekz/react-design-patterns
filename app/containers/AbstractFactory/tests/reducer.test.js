// import { fromJS } from 'immutable';
import abstractFactoryReducer from '../reducer';
import { DEFAULT_ACTION } from '../constants';

describe('abstractFactoryReducer', () => {
  it('returns the initial state', () => {
    expect(abstractFactoryReducer(undefined, {})).toMatchSnapshot();
  });

  it('returns the state when DEFAULT_ACTION is passed', () => {
    const action = {
      type: DEFAULT_ACTION,
    };

    expect(abstractFactoryReducer({}, action)).toMatchSnapshot();
  });
});
