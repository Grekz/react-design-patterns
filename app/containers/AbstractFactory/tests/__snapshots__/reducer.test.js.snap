// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`abstractFactoryReducer returns the initial state 1`] = `Immutable.Map {}`;

exports[`abstractFactoryReducer returns the state when DEFAULT_ACTION is passed 1`] = `Object {}`;
